src/lib/sql/sql_translator.cpp

```diff
@@ -355,11 +356,21 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_join(const hsql::Join
   auto left_node = _translate_table_ref(*join.left);
   auto right_node = _translate_table_ref(*join.right);
 
-  const hsql::Expr& condition = *join.condition;
+  std::vector<const hsql::Expr*> condition_list;
+  _flatten_conjunctive_expression_tree(join.condition, condition_list);
+
+  auto get_side =
+      std::bind(&SQLTranslator::_get_side, this, std::cref(left_node), std::cref(right_node), std::placeholders::_1);
+  std::vector<const hsql::Expr*> joins, left, right;
```

Comment:
those names don't mean anything, please name the variable so that they hint at what's in them.

Reply:
OK, possibly extending them with the suffix `_predicates`.